stream/distributedlog/core/src/main/java/org/apache/distributedlog/zk/ZKTransaction.java

```diff
@@ -63,13 +63,15 @@ public void addOp(Op<Object> operation) {
         if (!done.compareAndSet(false, true)) {
             return result;
         }
-        try {
-            zkc.get().multi(zkOps, this, result);
-        } catch (ZooKeeperClient.ZooKeeperConnectionException e) {
-            result.completeExceptionally(Utils.zkException(e, ""));
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            result.completeExceptionally(Utils.zkException(e, ""));
+        if (!zkOps.isEmpty()) {
+            try {
+                zkc.get().multi(zkOps, this, result);
```

Comment:
you are not completing the `CompletableFuture result`
please check 
`public void processResult(int rc, String path, Object ctx, List<OpResult> results) {`
at line 89